pkg/k8s/synced/resources.go

```diff
@@ -117,3 +143,59 @@ func (r *Resources) WaitForCacheSync(resourceNames ...string) {
 		}
 	}
 }
+
+// poll period for underlying client-go wait for cache sync.
+const syncedPollPeriod = 100 * time.Millisecond
+
+// WaitForCacheSyncWithTimeout waits for K8s resources represented by resourceNames to be synced.
+// For every resource type, if an event happens after starting the wait, the timeout will be pushed out
+// to be the time of the last event plus the timeout duration.
+func (r *Resources) WaitForCacheSyncWithTimeout(timeout time.Duration, resourceNames ...string) error {
+	defer func() {
+		r.Lock()
+		r.timeSinceLastEvent = nil
```

Comment:
Might help to leave a brief comment here on why we are doing this

Reply:
Sounds good, added comment, already squashed it into original commit to get branch ready.